### PR TITLE
share skill ctas

### PR DESCRIPTION
**Pull request by wingfeatherwave**
Screenshot 2024-12-03 at 2 26 29 PM
Screenshot 2024-12-03 at 2 20 02 PM
Screenshot 2024-12-03 at 1 00 25 PM
Screenshot 2024-12-03 at 2 16 19 PM

### CODE REVIEW

File: src/pages/GiveSkillNav.tsx [hunk 1]
```diff
@@ -44,6 +49,24 @@ export const GiveSkillNav = ({ skill }: { skill: string }) => {
       >
         <TimelineList fa size="lg" /> Leaderboard
       </Button>
+      <Button
+        color="textOnly"
+        onClick={() => setModalVisible(true)}
```
Comment (crabsinger) [inline src/pages/GiveSkillNav.tsx:54]:
```suggestion
        onClick={() => setModalVisible(prev => !prev)}
```

Using a function updater (prev => !prev) in the setState function is a good practice when updating state that depends on the previous state value. This ensures that the state update is based on the most recent state, which is particularly important when state updates are batched or if there are multiple state updates in quick succession. @wingfeatherwave

File: src/pages/GiveSkillNav.tsx [hunk 2]
```diff
@@ -44,6 +49,24 @@ export const GiveSkillNav = ({ skill }: { skill: string }) => {
...
+          ...tabStyles,
+        }}
+      >
+        <ShareGiveThick fa size="lg" /> Share
+      </Button>
+      {modalVisible && (
+        <DrawerModal
+          visible={modalVisible}
+          closeButtonStyles={{ color: '$white80' }}
+          onClose={() => setModalVisible(false)}
```
Comment (crabsinger) [inline src/pages/GiveSkillNav.tsx:65]: same

File: src/pages/colinks/give/ShareGiveCard.tsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        >
+          <ShareGiveContent skill={skill} />
+        </DrawerModal>
+      )}
+    </>
+  );
+};
+
+export const ShareGiveContent = ({ skill }: { skill?: string }) => {
+  const startPartyUrl = `https://warpcast.com/~/compose?text=https://give.party/${skill}&embeds[]=https://give.party/${skill}`;
```
Comment (crabsinger) [inline src/pages/colinks/give/ShareGiveCard.tsx:96]: i wonder if this works with this skill: https://coordinape.com/give/skill/%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA%20%F0%92%90%AA

haha

**Review by crabsinger — DISMISSED** [review]
@wingfeatherwave looks good, a few small nitty react things

File: src/pages/colinks/give/ShareGiveCard.tsx (hunk 3 above)
Comment (wingfeatherwave) [reply src/pages/colinks/give/ShareGiveCard.tsx:96]: 𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪𒐪

**Review by crabsinger — APPROVED** [review]

**Pull request merged**